doc/specs/stdlib_stats_distribution_exponential.md

```diff
@@ -0,0 +1,248 @@
+
+### Syntax
+
+`result = [[stdlib_stats_distribution_exponential(module):rvs_exp(interface)]]([lambda] [[, array_size]])`
+
+### Class
+
+Function
+
+### Arguments
+
+`lambda`: optional argument has `intent(in)` and is a scalar of type `real` or `complex`. The value of `lambda` has to be non-negative.
+
+`array_size`: optional argument has `intent(in)` and is a scalar of type `integer` with default kind.
+
+### Return value
+
+The result is a scalar or rank one array with a size of `array_size`, and as the same type of `lambda`.
+
+### Example
+
+```fortran
+program demo_exponential_rvs
+    use stdlib_random, only : random_seed
+    use stdlib_stats_distribution_exponential, only: rexp => rvs_exp
+
+    implicit none
+    real ::  a(2,3,4)
+    complex :: scale
+    integer :: seed_put, seed_get
+
+    seed_put = 1234567
+    call random_seed(seed_put, seed_get)
+
+    print *, rexp( )         !single standard exponential random variate
+
+! 0.358690143
+
+    print *, rexp(2.0)       !exponential random variate with lambda=2.0
+
+! 0.816459715
+
+    print *, rexp(0.3, 10)   !an array of 10 variates with lambda=0.3
+
+!  1.84008647E-02  3.59742008E-02  0.136567295  0.262772143  3.62352766E-02 
+!  0.547133625  0.213591918  4.10784185E-02  0.583882213  0.671128035
+
+    scale = (2.0, 0.7)
+    print *, rexp(scale)
+    !single complex exponential random variate with real part of lambda=2.0;
+    !imagainary part of lambda=0.7
+
+! (1.41435969,4.081114382E-02)
+
+end program demo_exponential_rvs
+```
+
+## `pdf_exp` - exponential distribution probability density function
+
+### Status
+
+Experimental
+
+### Description
+
+The probability density function (pdf) of the single real variable exponential distribution:
+
+$$f(x)=\begin{cases} \lambda e^{-\lambda x} &x\geqslant 0 \\\\ 0 &x< 0\end{}$$
+
+For a complex variable (x + y i) with independent real x and imaginary y parts, the joint probability density function
+is the product of the corresponding marginal pdf of real and imaginary pdf (for more details, see
+"Probability and Random Processes with Applications to Signal Processing and Communications", 2nd ed., Scott L. Miller and Donald Childers, 2012, p.197):
+
+$$f(x+\mathit{i}y)=f(x)f(y)=\begin{cases} \lambda_{x} \lambda_{y} e^{-(\lambda_{x} x + \lambda_{y} y)} &x\geqslant 0, y\geqslant 0 \\\\ 0 &otherwise\end{}$$
+
+### Syntax
+
+`result = [[stdlib_stats_distribution_exponential(module):pdf_exp(interface)]](x, lambda)`
+
+### Class
+
+Elemental function
+
+### Arguments
+
+`x`: has `intent(in)` and is a scalar of type `real` or `complex`.
+
+`lambda`: has `intent(in)` and is a scalar of type `real` or `complex`.
+
+All arguments must have the same type.
+
+### Return value
+
+The result is a scalar or an array, with a shape conformable to arguments, and as the same type of input arguments.
```

Comment:
Suggest to end this with "...arguments, with the same type as the input arguments."